lib/src/main/java/dev/openfeature/javasdk/IntegerHook.java

```diff
@@ -0,0 +1,9 @@
+package dev.openfeature.javasdk;
+
+public interface IntegerHook extends Hook<Integer> {
```

Comment:
I like this solution, I think it's a helpful guardrail, and it implies that if you use the generic, you have to be careful.